app/oapv_app_enc.c

```diff
@@ -753,6 +767,98 @@ static int update_param(args_var_t *vars, oapve_param_t *param)
     return 0;
 }
 
+static int parse_master_display(const char* data_string, oapvm_payload_mdcv_t *mdcv)
+{
+    int assigned_fields = sscanf(data_string,
+        "G(%u,%u)B(%u,%u)R(%u,%u)WP(%u,%u)L(%lu,%lu)",
+        &mdcv->primary_chromaticity_x[2], &mdcv->primary_chromaticity_y[2], // G
+        &mdcv->primary_chromaticity_x[1], &mdcv->primary_chromaticity_y[1], // B
+        &mdcv->primary_chromaticity_x[0], &mdcv->primary_chromaticity_y[0], // R
+        &mdcv->white_point_chromaticity_x, &mdcv->white_point_chromaticity_y, // White Point
+        &mdcv->max_mastering_luminance, &mdcv->min_mastering_luminance       // Luminance
+    );
+
+    // Check if sscanf successfully assigned all expected fields (10 numerical values).
+    const int expected_fields = 10;
+    if (assigned_fields != expected_fields) {
+        logerr("Parsing error: master diplay color volume information");
+        return -1;
+    }
+    return 0; // Success
+}
+
+static int parse_max_cll(const char* data_string, oapvm_payload_cll_t *cll)
+{
+    int assigned_fields = sscanf(data_string,
+        "%u,%u",
+        &cll->max_cll, &cll->max_fall
+    );
+
+    // Check if sscanf successfully assigned all expected fields (2 numerical values).
+    const int expected_fields = 2;
+    if (assigned_fields != expected_fields) {
+        logerr("ERR: parsing error: content light level information");
+        return -1;
+    }
+    return 0; // Success
+}
+
+static int update_metadata(args_var_t *vars, oapvm_t mid)
+{
+    int ret, size;
```

Comment:
ret is not initialized at the start of the function, so if everything succeeds, the function returns an undefined value. Please initialize ret to a default value (0 for success).